### PR TITLE
made the tally_vote fnc

### PR DESCRIPTION
**Pull request by Daheh**

### CODE REVIEW

**Comment by Daheh** [comment]
made tally_vote

File: worker.py [hunk 1]
```diff
@@ -7,28 +8,44 @@
...
+    r.hset('current_phrase', highestvote[0], int(nbWordPhrase))
+    r.delete('current_words')
+    #find the number of letter in our phrase bucket, if len > 1200
+    #we remove the letter exceding 1200
+    phrasewords = r.getall('current_phrase')
+    lenwords = int(r.hlen('current_phrase')-1)
+    for words in phrasewords:
+        lenwords = lenwords + len(words)
+    if lenwords > 1200:
+        return 'phrase too long'
```
Comment (mrrrgn) [inline worker.py]: Here you'll want to use an Exception. Instead of returning a string use: raise Exception("Phrase too long")

File: worker.py [hunk 2]
```diff
@@ -7,28 +8,44 @@
...
+    for i in range(0, bucket_capacity):
+        rndword = redis.StrictRedis.srandmember('wordbank')
+        redis.StrictRedis.rpush('current_words', rndword)
     """
     Pick words from a dictionary and dump them into a redis bucket.
     """
-
     # this return a random member of a set
     # redis.StrictRedis.srandmember(wordbank)
     pass
```
Comment (mrrrgn) [inline worker.py]: You'll want to remove the pass here, now that the function has content

File: worker.py [hunk 3]
```diff
@@ -7,28 +8,44 @@
 
 
 def tally_votes(max_phrase_length=1200):
+    #if we have a word with equal vote value, it will take the first one
+    votedwords = r.hgetall('current_words')
+    highestvote = []
+    highestvote.append('')
+    highestvote.append(0)
+    for y in votedwords.items():
```
Comment (mrrrgn) [inline worker.py]: The variable names here make this hatrd to follow. Also, python can handle looping over multiple arguments:
for thing_one, thing_two in dictionary.items():
    print(thing_one, thing_two)

File: worker.py [hunk 4]
```diff
@@ -7,28 +8,44 @@
 
 
 def tally_votes(max_phrase_length=1200):
+    #if we have a word with equal vote value, it will take the first one
+    votedwords = r.hgetall('current_words')
+    highestvote = []
```
Comment (mrrrgn) [inline worker.py]: Why is this a list?

File: worker.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env python
 import redis
+r = redis.StrictRedis(host='127.0.0.1', port=6379, db=0)
```
Comment (mrrrgn) [inline worker.py]: So, here and in the other places, we should make the host address, the port number, and the db number config variables. As in, store them in config.py and import them.

**Pull request merged**